File: Python/2.py
class Carro:
    def __init__(self, modelo, ano):
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0 
    
    def acelerar(self, aceleracao):
        self.velocidade += aceleracao
        print(f"O carro teve aceleração: {aceleracao} km/h.")
    
    def frear(self, desaceleracao):
        self.velocidade -= desaceleracao
        if self.velocidade < 0:
            self.velocidade = 0
        print(f"O carro freou: {desaceleracao} km/h.")
    
    def exibir_velocidade(self):
        print(f"Velocidade atual: {self.velocidade} km/h.")

carro = Carro("Strada", 2015)
carro.acelerar(80)  
carro.exibir_velocidade()  
carro.frear(40)  
carro.exibir_velocidade() 

File: Go/2/2.go
package main

import "fmt"

type Carro struct {
	velocidade int
}

func (c *Carro) Aceleracao(aumento int) {
	c.velocidade += aumento
}

func (c *Carro) Frenagem(reducao int) {
	c.velocidade -= reducao
	if c.velocidade < 0 {
		c.velocidade = 0
	}
}

func (c *Carro) ExibirVelocidade() {
	fmt.Printf("A velocidade atual é: %d km/h\n", c.velocidade)
}

func main() {

	carro := Carro{velocidade: 0}

	carro.Aceleracao(80)
	carro.ExibirVelocidade()

	carro.Frenagem(20)
	carro.ExibirVelocidade()

	carro.Frenagem(15)
	carro.ExibirVelocidade()
}
